File: Basic Programming/Hackerrank 30 days of code c++ solution/Day_8.cpp
/*                    Name : Arif Khan
                      Judge: HACKERRANK
                      University: Primeasia University
                      problem: Day 8: Dictionaries and Maps
                      Difficulty: Easy
                      Problem Link: https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem
*/

#include <map>
#include <iostream>
using namespace std;


int main() {
    map<string,int>mymap;
  int n,k;
    string name,s_name;
    cin>>n;
    while(n--){
        cin>>name>>k;
        mymap.insert(pair<string,int>(name,k));
    }
    while(cin>>s_name){
        
             if(mymap.count(s_name) != 0)
                 cout<<mymap.find(s_name)->first<<"="<<mymap.find(s_name)->second<<endl;
            else
                 cout<<"Not found"<<endl;
    }
    return 0;
}

File: Basic Programming/Hackerrank 30 days of code python3 solution/Day_8.py
#                      Name : Arif Khan
#                      Judge: HACKERRANK
#                      University: Primeasia University
#                      problem: Day 8: Dictionaries and Maps
#                      Difficulty: Easy
#                      Problem Link: https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem
#

from sys import stdin

if __name__ == "__main__":
    lines = stdin.read().splitlines()
    
    databook = dict(i.split() for i in lines[1:int(lines[0])+1])
    
    print('\n'.join(["{0:}={1:}".format(i,databook[i]) if i in databook else "Not found" for i in lines[int(lines[0])+1:]]))